From 1cb79e7793c1a556ac999b87360303fa370ce323 Mon Sep 17 00:00:00 2001
From: Yuvasree Sivasankaran <quic_ysivasan@quicinc.com>
Date: Fri, 19 Jan 2024 00:17:46 +0530
Subject: [PATCH] wifi: mac80211: Avoid linkid WARN_ON during recovery

During FW recovery, first vdev's get starts followed
by peer create. In case of AP+Monitor, monitor start
capturing before peer create completion but mac80211
have associated sta entry which leads to warn_on.
Add in_reconfig flag check for warn_on which points the
recovery state.

Signed-off-by: Yuvasree Sivasankaran <quic_ysivasan@quicinc.com>
---
 net/mac80211/rx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net/mac80211/rx.c b/net/mac80211/rx.c
index f08a557..94c7dbb 100644
--- a/net/mac80211/rx.c
+++ b/net/mac80211/rx.c
@@ -4455,7 +4455,8 @@ static bool ieee80211_rx_data_set_sta(struct ieee80211_rx_data *rx,
 				rx->link = rcu_dereference(rx->sdata->link[sta->deflink.link_id]);
 			} else {
 				rx->link = &rx->sdata->deflink;
-				WARN_ON_ONCE(1);
+				if (!rx->local->in_reconfig)
+					WARN_ON_ONCE(1);
 			}
 		} else {
 			rx->link = &rx->sdata->deflink;
-- 
2.34.1

